core/client/src/main/java/alluxio/client/netty/NettyRemoteBlockReader.java

```diff
@@ -58,13 +58,11 @@ public NettyRemoteBlockReader() {
   @Override
   public ByteBuffer readRemoteBlock(InetSocketAddress address, long blockId, long offset,
       long length, long lockId, long sessionId) throws IOException {
-
+    SingleResponseListener listener = new SingleResponseListener();
```

Comment:
I'd rather not call `mHandler.removeListener(listener)` when we haven't yet called `mHandler.addListener` (this could happen if an exception occurs). Could we set this `null` here, initialize it inside the `try`, and only do the remove if it isn't `null`? Alternately we could nest another `try ... finally`.